### DISCUSSION TITLE
Helix symbol search for Rust

### DISCUSSION THREAD

**Discussion by blitzerr** [helix-editor/helix#5433] [General] [answered]
I have a couple of questions:
1. How to I navigate to a symbol definition vs implementation vs all references for the symbol currently under cursor ?
2. For a very large project, how to I search across all symbols. I see that the helix dialog for global synbols, that I get by `spc` + `S`(capital S) only shows 121 symbols. The project is huge, so I am very sure there are more than that number. Do I need to set something up in rust-analyzer ? Here also, how do I paste the copied string into the search bar of the dialog that opended.

**Comment by Ordoviz** [comment, marked as answer]
For the first question: Press `gd`, `gi`, or `gr`, respectively.

**Reply by blitzerr** [reply]
The second part is answered by @gabydd in the following [one](https://github.com/helix-editor/helix/discussions/5433#discussioncomment-4619692). GH won't let me select two answers to a question.

**Reply by gabydd** [reply]
Just fyi if you use the install command in the readme add ``--locked`` cause there is a dependency issue

**Reply by blitzerr** [reply]
Awesome! Thanks for the heads up

**Comment by gabydd** [comment]
For the second one language servers will not send all the symbols at once so you will have to search a bit to get more symbols, note this will only work if you are using the master branch, see https://github.com/helix-editor/helix/issues/1437 where it also shows how to increase the limit of symbols that rust analyzer will send

**Reply by blitzerr** [reply]
I guess, then, when is the next release of Helix ? :)

**Reply by gabydd** [reply]
See the FAQ https://github.com/helix-editor/helix/issues/1437 generally there will be a new release every couple months, but all time is volunteered so no specifics can be made. If you want to use a feature, you can build from master(if you don't want to build it yourself if you are on arch there is a helix-git package)

**Reply by blitzerr** [reply]
I will build it from the master. I was just worried I will also build some other bugs along with this :)

**Reply by kirawi** [reply]
Master is generally pretty stable. Most devs run it.

**Comment by blitzerr** [comment]
Thanks a lot @gabydd  and @Ordoviz for your answer. Together they answer both my questions. This is an excellent community. I always get answers to my questions here. Such helpful people. You all are awesome !
